src/metabase/lib/metadata/protocols.cljc

```diff
@@ -1,11 +1,10 @@
 (ns metabase.lib.metadata.protocols
   (:require
-   [metabase.shared.util.i18n :as i18n]
-   #?@(:clj
-       ([potemkin :as p]
-        [pretty.core :as pretty]))))
+   [medley.core :as m]
+   #?@(:clj ([potemkin :as p]
+             [pretty.core :as pretty]))))
 
-(#?(:clj p/defprotocol+ :cljs defprotocol) DatabaseMetadataProvider
+(#?(:clj p/defprotocol+ :cljs defprotocol) MetadataProvider
```

Comment:
I reworked the `DatabaseMetadataProvider` protocol to include a lot more stuff that was on the TODO-list, for example methods to fetch specific tables by ID and other things that we're going to need eventually when we implement a QP store version of the provider or try to hook it in to the Redux store if we end up doing that 

I added methods for Metrics and Segments too which we will need sooner rather than later. `Metadata.ts` already has them. Since this isn't just DB stuff anymore I renamed it to `MetadataProvider`